scripts/randomx_booster.sh

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env bash
+# based on xmrig's randomx_boost.sh script
+# lifted by psychocrypt
+
+function help()
+{
+    echo "$(basename $0) modifies caching behaviors of your CPU"
+    echo "and activates huge pages."
+    echo "Reboot your system to revert the changes."
+    echo ""
+    echo "must be called with administrative privileges e.g. 'sudo $(basename $0)'"
+}
+
+if [ $# -ge 1 ] ; then
+    help
+    exit 1
+fi
+
+hasAptGet=$(which apt-get >/dev/null && { echo 1; } || { echo 0; })
+hasApt=$(which apt >/dev/null && { echo 1; } || { echo 0; })
+hasYum=$(which yum >/dev/null && { echo 1; } || { echo 0; })
+
+tools=$(which wrmsr >/dev/null || { echo "msr-tools "; })$(which numactl >/dev/null || { echo " numactl"; })
+
+if [ -n "$tools" ] ; then
+    echo "tool '$tools' not found, $(basename $0) is trying to install the dependency"
+    if [ $hasAptGet -eq 1 ] ; then
+        comm="apt-get --no-install-recommends --yes install $tools"
+        echo "execute: $comm"
+        $comm
+    elif [ $hasApt -eq 1 ] ; then
+        comm="apt-get --no-install-recommends --yes install $tools"
+        echo "execute: $comm"
+        $comm
+    elif [ $hasYum -eq 1 ] ; then
+        comm="yum install -y $tools"
+        echo "execute: $comm"
+        $comm
+    else
+        echo "package manager unknown, please install '$tools' by hand" >&2
+        exit 1
+    fi
+fi
+
+hasWrmsr=$(which wrmsr >/dev/null && { echo 1; } || { echo 0; })
+if [ $hasWrmsr -eq 0 ] ; then
+    echo "dependency 'wrmsr' not found, tool failed" >&2
+    exit 1
+fi
+
+hasNumactl=$(which numactl >/dev/null && { echo 1; } || { echo 0; })
+if [ $hasNumactl -eq 0 ] ; then
+    echo "dependency 'numactl' not found, tool failed" >&2
+    exit 1
+fi
+
+echo ""
+modprobe msr
+
+if cat /proc/cpuinfo | grep -q "AMD Ryzen" ;
+	then
+		echo "Detected Ryzen"
+		wrmsr -a 0xc0011022 0x510000
+		wrmsr -a 0xc001102b 0x1808cc16
+		wrmsr -a 0xc0011020 0
+		wrmsr -a 0xc0011021 0x40
+		echo "MSR register values for Ryzen applied"
```

Comment:
Non blocking comment - those numbers are correct only for gen1 zen cores. Some testing is needed as they will set collateral bits with unknown function on gen2.

Reply:
Numbers for C001_1020 and C001_1021 registers were taken from R7 3700X (Geekbench performance bias option in BIOS). Funny to see how you copy my work without any clue of what it does or how it was obtained :D AMD doesn't change MSR registers (at least these ones) within same family CPUs, and all Zen, Zen+ and Zen2 CPUs so far are family 17h.

Reply:
> Numbers for C001_1020 and C001_1021 registers were taken from R7 3700X (Geekbench performance bias option in BIOS). 
> AMD doesn't change MSR registers (at least these ones) within same family CPUs, and all Zen, Zen+ and Zen2 CPUs so far are family 17h.

+1, for the info

> Funny to see how you copy my work without any clue of what it does or how it was obtained :D 

please stop craying

Reply:
> Funny to see how you copy my work without any clue of what it does or how it was obtained :D 

Of course I know - you copied it from RagerX. That's why you have 2 gen2 MSRs and 2 gen1 MSRs. If you don't stop rocking your arse off, we will ban you from this repo.

Reply:
OMG lol. I figured the last two values myself, and the first two were sent to me by xnbya who tested prefetcher options in BIOS on some other AM4 board (that I didn't have). Your RagerX is a joke. Have you even implemented MSR mod there yet?

Reply:
Go ahead and ban me, it's will be such a stamp of approval :D

Reply:
> stamp of approval _(by monero)_
yay, thats the moment of truth right here xD what actually for are you ranting 24/7 and blame stak team in everything.
btw, how is your [viruses claim doing](https://www.reddit.com/r/XmrStak/comments/dxrlgx/warning_according_to_randomx_dev_xmrstakrx_source/)? found any so far? it is a month old claim already..

Reply:
FYI this is the screenshot of #monero-pools IRC from December 11th (everyone with IRC logs can check it): https://imgur.com/a/cE5XqHE
This where I got the info about the first 2 MSR registers and then I figured out the other two on my own, testing my BIOS.

Reply:
RagerX came out on the 9th -> see, it only works on people who are dumb enough not to fact-check you.

Reply:
Yep, it came out on the 9th and it didn't have MSR mod back then because I tested it and I didn't see anything faster than I already had with xmrig on windows (without the mod). Maybe It has MSR mod now (of course), but I don't really care.

Reply:
It's very convenient to claim you had something before XMRig without proof when everything RagerX is closed source and people have no way of checking it. You're a pathological liar.

Reply:
You have been hanging out with pathological liars a bit too much.

https://medium.com/@crypto_ryo/senior-nasa-engineer-calls-out-howard-chu-hyc-symas-monero-developer-on-being-a-fraud-db0b46688041

But, hey, if "Look! I got this magic undocumented number of some guy on IRC. I totally don't know where it came from!!" is your story, that's your story.

Follow up question - do you think we would be dumb enough to put all our improvements at once :P. You will need to start looking harder for dodgy guys on IRC to find stuff for you =]